pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -122,15 +134,18 @@ func (l *LeaderElection) AddRunFunction(f func(stop <-chan struct{})) *LeaderEle
 	return l
 }
 
-func NewLeaderElection(namespace, name, electionID string, client kubernetes.Interface) *LeaderElection {
+func NewLeaderElection(namespace, name, electionID, revision string, client kube.Client) *LeaderElection {
 	if name == "" {
 		name = "unknown"
 	}
+	watcher := revisions.NewDefaultWatcher(client, revision, make(chan struct{}))
```

Comment:
Maybe  `s.environment.DefaultWatcher`should be used instead of making multiple

Reply:
Yeah agreed, had it like this to experiment with Go-routine usage, will clean up and move out of draft.